src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -1421,4 +1415,93 @@ private void OnSpeciesNameTextChanged(string newText)
             speciesNameEdit.Set("custom_colors/font_color", new Color(1, 1, 1));
         }
     }
+    private void createAndUpdateProcessBar (KeyValuePair <string, float> process, ProgressBar parent)
+    {
+        if (parent.HasNode(process.Key))
+        {
+            IconProgressBar progressBar = (IconProgressBar) parent.GetNode(process.Key);
+            TextureRect textureRect = progressBar.GetChild<TextureRect>(0);
+            progressBar.MarginLeft = getPreviousBar(parent, progressBar).RectSize.x + getPreviousBar(parent, progressBar).MarginLeft;
+            progressBar.RectSize = new Vector2((float)Math.Floor(process.Value / parent.MaxValue * 318), 30);
+            textureRect.Visible = progressBar.RectSize.x >= 30;
+        }
+        else
+        {
+            IconProgressBar progressBar = (IconProgressBar) IconProgressBarScene.Instance();
+            progressBar.Name = process.Key;
+            progressBar.Color = ATPBarHelper.GetBarColour(process.Key, parent.Name);
+            TextureRect textureRect = progressBar.GetChild<TextureRect>(0);
+            textureRect.Texture = ATPBarHelper.GetBarIcon(process.Key);
+            textureRect.Expand = true;
+            textureRect.RectSize = new Vector2(30, 30);
+            parent.AddChild(progressBar);
+            progressBar.MarginLeft = getPreviousBar(parent, progressBar).RectSize.x + getPreviousBar(parent, progressBar).MarginLeft;
+            progressBar.RectSize = new Vector2((float)Math.Floor(process.Value / parent.MaxValue * 318), 30);
+            textureRect.Visible = progressBar.RectSize.x >= 30;
+            progressBar.Connect("gui_input", this, nameof(atpBarToggled), new Godot.Collections.Array(){progressBar});
+        }
+    }
+
+    private void atpBarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            bar.disabled = !bar.disabled;
+            handleATPBarDisabling(bar, eventMouse);
+        }
+    }
+
+    private void handleATPBarDisabling(IconProgressBar bar, InputEventMouseButton eventMouse)
+    {
+        if (bar.disabled)
+        {
+            bar.GetChild<TextureRect>(0).Modulate = new Color("#000000");
```

Comment:
I think that this logic should also be in the icon bar class, make like a new property Disabled which sets and unsets the modulate colours.